pkg/compactor/blocks_cleaner.go

```diff
@@ -571,22 +571,32 @@ func listBlocksOutsideRetentionPeriod(idx *bucketindex.Index, threshold time.Tim
 	return
 }
 
-// findMostRecentModifiedTimeForBlock finds the most recent modification time for all files in a block.
-func findMostRecentModifiedTimeForBlock(ctx context.Context, blockID ulid.ULID, userBucket objstore.Bucket) (time.Time, error) {
-	var result time.Time
+var errStopIter = errors.New("stop iteration")
 
-	err := userBucket.Iter(ctx, blockID.String(), func(name string) error {
+// stalePartialBlockLastModifiedTime returns the most recent last modified time of a stale partial block, or the zero value of time.Time if the provided block wasn't a stale partial block
+func stalePartialBlockLastModifiedTime(ctx context.Context, blockID ulid.ULID, userBucket objstore.InstrumentedBucket, partialDeletionCutoffTime time.Time) (time.Time, error) {
+	var lastModified time.Time
+	err := userBucket.WithExpectedErrs(func(err error) bool {
+		return errors.Is(err, errStopIter) // sentinel error
+	}).Iter(ctx, blockID.String(), func(name string) error {
 		if strings.HasSuffix(name, objstore.DirDelim) {
 			return nil
 		}
 		attrib, err := userBucket.Attributes(ctx, name)
 		if err != nil {
 			return errors.Wrapf(err, "failed to get attributes for %s", name)
 		}
-		if attrib.LastModified.After(result) {
-			result = attrib.LastModified
+		if !attrib.LastModified.Before(partialDeletionCutoffTime) {
```

Comment:
Tiny simplification:

```suggestion
		if attrib.LastModified.After(partialDeletionCutoffTime) {
```

Reply:
I like this, changed the test as well to reflect it the equality case changing. The documentation wasn't specific about this and the difference never matters in practice.